### PR TITLE
(GH-2910) Relativize  bolt paths

### PR DESCRIPTION
**Pull request by artlawson**
Use `require_relative` in place of `require` for Bolt performance improvements.

!no-release-note

### CODE REVIEW

**Review by beechtom — CHANGES_REQUESTED** [review]
The paths in the `bolt_spec` and `bolt_server` files are incorrect. I think we should avoid changing these anyways and focus on just `bolt` files (unless @lucywyman has other thoughts?).

**Review by beechtom — CHANGES_REQUESTED** [review]
This needs to be rebased off `main`.

**Comment by donoghuc** [comment]
I would love it if somebody builds this into a PE install and runs the orch tests against it. Given the issues we saw with puppet moving to relative_require i want to double check there wont be strange code loading issues.

**Comment by lucywyman** [comment]
Sweet, finally got a green run for this in the Orchestrator!

File: lib/bolt/secret.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
```
Comment (beechtom) [inline lib/bolt/secret.rb:1]: Need to delete this file.

**Review by beechtom — CHANGES_REQUESTED** [review]
1. This also needs to update paths in `lib/bolt/application.rb`.
2. The commit message has two release note tags. We should probably remove `!feature`.

**Review by beechtom — APPROVED** [review]

**Pull request merged**